src/Razor/Microsoft.AspNetCore.Razor.Language/src/CodeGeneration/CodeWriterExtensions.cs

```diff
@@ -145,6 +145,27 @@ public static CodeWriter WriteUsing(this CodeWriter writer, string name, bool en
             return writer;
         }
 
+        public static CodeWriter WriteEnhancedLineNumberDirective(this CodeWriter writer, SourceSpan span, int characterOffset)
```

Comment:
Any reason to not put this in the line pragma writer class?

Reply:
The `LinePragmaWriter` class currently calls out to the `WriteLineNumberDirective` function which writes standard line pragmas. I figured I'd follow the convention and use a separate method to write the enhanced pragmas. This made more sense to me than creating a new `EnhancedLinePragmaWriter` or inlining the codewriting. Let me know if I misunderstood your recommendation.

Reply:
Does anything else call the `WriteLineNumberDIrective`? Assuming not it's probably just a mistake on our part 😄. Otherwise we'll be encouraging folks to render line pragmas independently which doesn't actually generate source mappings which is concerning

Reply:
Nope -- it's only ever invoked in the `LinePramgaWriter`.